### PR TITLE
Use scriv for CHANGELOG generation in release workflow

### PR DESCRIPTION
**Pull request by jwodder**
Part of #6918.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -28,34 +28,29 @@ jobs:
...
-          elif ! dpkg --compare-versions "$new_version" gt "$changelog_version"
-          then echo "[ERROR] Could not compare new version '$new_version' to changelog version '$changelog_version'"
+          if [ "$auto_version" != "$changelog_version" ]
+          then echo "[ERROR] Auto version does not equal changelog version"
                exit 1
           fi
-          ~/auto shipit -vv $opts
+          scriv collect --add --version "$changelog_version"
+          git commit -m '[skip ci] Update CHANGELOG' CHANGELOG.md
+          ~/auto shipit -vv --no-changelog
```
Comment (yarikoptic) [inline .github/workflows/release.yml:53]: if `--no-changelog` -- would it trigger 
```
.autorc:                "afterChangelog": "make update-changelog && git add docs/source/changelog.rst && git commit -m '[skip ci] Update RST changelog'",
```
?  may be better to just move that call right here before `shipit`?

File: changelog.d/templates/entry_title.md.j2 [hunk 2]
```diff
@@ -0,0 +1 @@
+{{ version if version else "VERSION" }} ({{ date.strftime('%Y-%m-%d') }})
```
Comment (yarikoptic) [inline changelog.d/templates/entry_title.md.j2:1]: when would that `"VERSION"` happen? should we just error may be to not end up with some "broken" changelog?

**Review by yarikoptic — COMMENTED** [review]
A few minor questions/comments. Otherwise -- looks worth trying! Thank you!

File: changelog.d/templates/entry_title.md.j2 (hunk 2 above)
Comment (jwodder) [reply changelog.d/templates/entry_title.md.j2:1]: With this setup, I don't believe it'll happen; I just copied the template from datalad-next.

File: .github/workflows/release.yml (hunk 1 above)
Comment (jwodder) [reply .github/workflows/release.yml:55]: Done.

**Comment by yarikoptic** [comment]
ok, and with this we will test if it all works -- added a `release` label. Thank you @jwodder

**Comment by yarikoptic-gitmate** [comment]
PR released in [`ignore-me-testing`](https://github.com/datalad/datalad/releases/tag/ignore-me-testing)

**Comment by yarikoptic-gitmate** [comment]
PR released in [`0.17.6`](https://github.com/datalad/datalad/releases/tag/0.17.6)

**Pull request merged**